### PR TITLE
Test cleanup around onPushReceived

### PR DESCRIPTION
**Pull request by jkasten2**
* Tests for ServiceWorkerManager.ts should test only the usage from a page.
* The new context/sw tests should be for the ServiceWorker itself.

---
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-website-sdk/599)

### CODE REVIEW

File: src/service-worker/ServiceWorker.ts [hunk 1]
```diff
@@ -202,11 +196,11 @@ export class ServiceWorker {
                 // Never nest the following line in a callback from the point of entering from retrieveNotifications
                 notificationEventPromiseFns.push((notif => {
                   return ServiceWorker.displayNotification(notif)
-                      .then(() => ServiceWorker.updateBackupNotification(notif).catch(e => Log.error(e)))
                       .then(() => {
                         return ServiceWorker.workerMessenger.broadcast(WorkerMessengerCommand.NotificationDisplayed, notif).catch(e => Log.error(e))
```
Comment (rgomezp) [inline src/service-worker/ServiceWorker.ts]: Linter complains about line length

File: src/service-worker/ServiceWorker.ts [hunk 2]
```diff
@@ -202,11 +196,11 @@ export class ServiceWorker {
                 // Never nest the following line in a callback from the point of entering from retrieveNotifications
                 notificationEventPromiseFns.push((notif => {
                   return ServiceWorker.displayNotification(notif)
```
Comment (rgomezp) [inline src/service-worker/ServiceWorker.ts]: Is `return` necessary here?

File: src/services/Database.ts [hunk 3]
```diff
@@ -262,8 +261,6 @@ export default class Database {
       await this.put("Ids", {type: "WORKER1_ONE_SIGNAL_SW_VERSION", id: state.workerVersion});
     if (state.updaterWorkerVersion)
       await this.put("Ids", {type: "WORKER2_ONE_SIGNAL_SW_VERSION", id: state.updaterWorkerVersion});
-    if (state.backupNotification)
-      await this.put("Ids", {type: "backupNotification", id: state.backupNotification});
```
Comment (rgomezp) [inline src/services/Database.ts]: Nit: there should be a space after and before brackets

Comment (rgomezp) [reply src/services/Database.ts]: same for above calls

**Comment by rgomezp** [comment]
LGTM except a couple nits. After, I think it's good to merge

File: src/service-worker/ServiceWorker.ts (hunk 1 above)
Comment (jkasten2) [reply src/service-worker/ServiceWorker.ts]: This line wasn't changed in this PR, thinking we should leave as is for now as this whole method needs a refactor

File: src/services/Database.ts (hunk 3 above)
Comment (jkasten2) [reply src/services/Database.ts]: 👍

**Pull request merged**